tests/test_utils.py

```diff
@@ -33,4 +33,31 @@ def test_symeig_custom_low_rank():
     # symeig should fail for low-rank
     assert not torch.all(l1 >= 0.0)
     # test clamping to zeros
-    assert torch.all(l2 >= 0.0)
\ No newline at end of file
+    assert torch.all(l2 >= 0.0)
+
+    
+def test_diagonal_normal_samples():
```

Comment:
Is it possible to test against torch `Normal`/`MultivariateNormal` samples by using the same generator state or is the implementation not the same?

Reply:
Checked it, unfortunately not. But the statistics match when taking many samples.